### ISSUE TITLE
[Bug]: Provider crashes when creating a view with a masking policy

### ISSUE THREAD

**Issue by algrice** [Snowflake-Labs/terraform-provider-snowflake#3219]
### Terraform CLI Version

1.9.3

### Terraform Provider Version

0.96.0

### Terraform Configuration

```terraform
resource "snowflake_view" "reproduce_error" {
  database        = "EXAMPLE_DATABASE"
  schema          = "EXAMPLE_SCHEMA"
  name            = upper("test_view")
  is_secure       = false
  change_tracking = false
  is_temporary    = false

  column {
    column_name = "id"
  }

  column {
    column_name = "version"
  }

  column {
    column_name = "data_source"
    masking_policy {
      policy_name = snowflake_masking_policy.test_policy.fully_qualified_name
      using       = ["data_source"]
    }
  }

  statement = <<-EOF
SELECT id, version, data_source
FROM "EXAMPLE_DATABASE"."EXAMPLE_SCHEMA"."EXAMPLETABLE";
EOF
}
```

### Category

category:resource

### Object type(s)

resource:view

### Expected Behavior

I expected the view to get created successfully and have the masking policy applied to the column with no errors. 

### Actual Behavior

The view _did_ get created successfully in Snowflake, but the terraform provider crashed immediately afterwards. 

### Steps to Reproduce

Copy the terraform from above after replacing the database / schema / query and policy reference with real resources. Run terraform apply.  

### How much impact is this issue causing?

Medium

### Logs

https://gist.github.com/algrice/da65a8bd4f5ce5bc120609c006650241

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-jmichalak** [comment]
Hi @algrice 👋 

We've released fixes for `view` resource in v0.97 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.97.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0960--v0970)). Please upgrade to this version. I've verified this now, and in v097 a similar setup works correctly. If not, please let us know. In this case, please provide more logs before the crash, with `TF_LOG=DEBUG`.

**Comment by algrice** [comment]
> We've released fixes for `view` resource in v0.97 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.97.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0960--v0970)). Please upgrade to this version. I've verified this now, and in v097 a similar setup works correctly. If not, please let us know. In this case, please provide more logs before the crash, with `TF_LOG=DEBUG`.

Hi! Thanks for the suggestion. Unfortunately we haven't been able to test anything greater than v0.96 because we're blocked on this issue https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/3136

We can continue working around by creating views with a `snowflake_unsafe_execute` resource unless you have a better work around?

**Comment by sfc-gh-jcieslak** [comment]
Hey @algrice 
If that helps, #3136 will be fixed (along with many other task issues) in the provider's next (v0.99.0) version.

**Comment by algrice** [comment]
Hi! We've been able to upgrade to v0.99 and can confirm we can now create views with masking policies using the snowflake_view resource. 😃 

One minor bug we're seeing on v0.99: if we don't pass a `using` on the `masking_policy` then we see the provider crash again despite that field being documented as optional. https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/view#using-1 

That's not at all a blocker for us though. Thanks!

**Comment by sfc-gh-jmichalak** [comment]
Hi @algrice 👋 

We've just released v0.100.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.100.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0990--v01000)) with the fix. Please upgrade to this version and let us know if you have any issues, preferably in a new issue.
